core/src/test/java/co/cask/wrangler/statistics/ZipcodeTest.java

```diff
@@ -0,0 +1,125 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.statistics;
+
+import au.com.bytecode.opencsv.CSVReader;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.wrangler.api.Record;
+import co.cask.wrangler.api.statistics.Statistics;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Test zip code regex detecting. See {@link BasicStatistics} for Regex pattern definitions
+ */
+public class ZipcodeTest {
+  private static final String DATA_FILE = "MOCK_DATA.csv";
+
+  @Test
+  public void testZipcode() throws Exception {
+
+    Record mock1 = new Record();
+
+    //US and Mexico
+    mock1.add("zip_code_1", "94105-0011");
+    mock1.add("zip_code_2", "61801");
+    mock1.add("zip_code_3", "61820");
+
+    //China and India
+    mock1.add("zip_code_4", "030002");
+    mock1.add("zip_code_5", "030001");
+    mock1.add("zip_code_6", "382355");
+
+    //Canada
+    mock1.add("zip_code_7", "V6T 1Z4");
+    mock1.add("zip_code_8", "N2L 3G1");
+
+    //no use
+    List<Record> records = Arrays.asList(
+            mock1, mock1, mock1
+    );
+    Statistics statisticsGen = new BasicStatistics();
+    Record summary = statisticsGen.aggregate(records);
+    Record typesList = (Record) summary.getValue("types");
+
+    for (int i = 0; i < typesList.length(); i ++) {
+      ArrayList<KeyValue> types = (ArrayList<KeyValue>) typesList.getValue(i);
+      Assert.assertEquals(types.size(), 1);
+      Assert.assertEquals(types.get(0).getKey(), "Zip_Code");
+    }
+  }
+
+  @Test
+  public void testNonZipcode() throws Exception {
+
+    List<Record> records = new ArrayList<>();
+
+    List<String> cardNumbers = readCreditCardFromCsv(DATA_FILE);
+    for (String cardNumber : cardNumbers) {
+      Record record = new Record();
+      record.add("card_number", cardNumber);
+      records.add(record);
+    }
+
+    Statistics statisticsGen = new BasicStatistics();
+    Record summary = statisticsGen.aggregate(records);
+    Record typesList = (Record) summary.getValue("types");
+
+    for (int i = 0; i < typesList.length(); i ++) {
+      ArrayList<KeyValue> types = (ArrayList<KeyValue>) typesList.getValue(i);
+      for (KeyValue keyValue : types) {
+        /*
+        Assert.assertFalse(keyValue.getKey().equals("CN_Code"));
+        Assert.assertFalse(keyValue.getKey().equals("IN_Code"));
+        Assert.assertFalse(keyValue.getKey().equals("US_Code"));
+        Assert.assertFalse(keyValue.getKey().equals("MX_Code"));
+        Assert.assertFalse(keyValue.getKey().equals("CA_Code"));
+        */
+        Assert.assertFalse(keyValue.getKey().equals("Zip_Code"));
+      }
+    }
+  }
+
+
+  private List<String> readCreditCardFromCsv(String fileName) {
```

Comment:
They are not the exactly same function, and need different logs. Maybe is't better to define them separately?